src/TLuaInterpreter.cpp

```diff
@@ -1109,43 +1109,43 @@ int TLuaInterpreter::addMapEvent(lua_State * L){
     QStringList actionInfo;
     if( ! lua_isstring( L, 1 ) )
     {
-        lua_pushstring( L, "addMapEvent: wrong first argument type" );
-        lua_error( L );
+        lua_pushfstring(L, "addMapEvent: bad argument #1 type (uniquename as string expected, got %s!)", luaL_typename(L, 1));
+        lua_error(L);
         return 1;
     }
     else
     {
-        uniqueName = lua_tostring( L, 1 );
+        uniqueName = QString::fromUtf8( lua_tostring( L, 1 ) );
     }
     if( ! lua_isstring( L, 2 ) )
     {
-        lua_pushstring( L, "addMapEvent: wrong second argument type" );
-        lua_error( L );
+        lua_pushfstring(L, "addMapEvent: bad argument #2 type (event name as string expected, got %s!)", luaL_typename(L, 1));
+        lua_error(L);
         return 1;
     }
     else
     {
-        actionInfo << lua_tostring( L, 2 );
+        actionInfo << QString::fromUtf8( lua_tostring( L, 2 ) );
     }
     if( ! lua_isstring( L, 3 ) )
     {
         actionInfo << "";
     }
     else
     {
-        actionInfo << lua_tostring( L, 3 );
+        actionInfo << QString::fromUtf8( lua_tostring( L, 3 ) );
     }
     if( ! lua_isstring( L, 4 ) )
     {
         actionInfo << uniqueName;
     }
     else
     {
-        actionInfo << lua_tostring( L, 4 );
+        actionInfo << QString::fromUtf8( lua_tostring( L, 4 ) );
     }
     //variable number of arguments
     for (int i=5;i<=lua_gettop(L);i++)
```

Comment:
Won't the absence of the `{`...`}` around the following line clash with the already approved #1115?